scripts/Packages.scala

```diff
@@ -13,27 +13,86 @@ import scala.util.control.NonFatal
-      case "publish-maven" :: tail => publishMavenSelected(generatedFile, tail)
-      case _ => println(s"Unknown command: $args, expected one of: " +
-        s"metadata-all, metadata, generate-all, generate, publish-local-all, publish-local, publish-maven-all, publish-maven")
+      case "metadata-all" :: Nil       => downloadPackagesMetadata(packagesDir)
+      case "metadata" :: tail          => downloadPackagesMetadata(packagesDir, selected = tail)
+      case "generate-all" :: Nil       => generateAll(packagesDir)
+      case "generate" :: tail          => generateSelected(packagesDir, tail)
+      case "publish-local-all" :: Nil  => publishLocalAll(generatedFile)
+      case "publish-local" :: tail     => publishLocalSelected(generatedFile, tail)
+      case "publish-maven-all" :: Nil  => publishMavenAll(generatedFile)
+      case "publish-maven" :: tail     => publishMavenSelected(generatedFile, tail)
+      case "publish-github-all" :: Nil => publishGithubAll(generatedFile)
+      case "publish-github" :: tail    => publishGithubSelected(generatedFile, tail)
+      case _ =>
+        println(
+          s"""Unknown command: $args, expected one of:
+             |  metadata-all                - download all packages metadata
+             |  metadata <package>...       - download selected packages metadata
+             |  generate-all                - generate all packages source code
+             |  generate <package>...       - generate selected packages source code
+             |  publish-local-all           - publish all packages locally
+             |  publish-local <package>...  - publish selected packages locally
+             |  publish-maven-all           - publish all packages to Maven
+             |  publish-maven <package>...  - publish selected packages to Maven
+             |  publish-github-all          - publish all packages to Github
+             |  publish-github <package>... - publish selected packages to Github
+             |""".stripMargin
+        )
   end main
 
-  val cwd = besomDir
-  val codegenDir = cwd / ".out" / "codegen"
-  val packagesDir = cwd / ".out" / "packages"
-  val publishLocalDir = cwd / ".out" / "publishLocal"
-  val publishMavenDir = cwd / ".out" / "publishMaven"
+  val cwd              = besomDir
+  val codegenDir       = cwd / ".out" / "codegen"
+  val packagesDir      = cwd / ".out" / "packages"
+  val publishLocalDir  = cwd / ".out" / "publishLocal"
+  val publishMavenDir  = cwd / ".out" / "publishMaven"
+  val publishGithubDir = cwd / ".out" / "publishGithub"
+
+  val generatedFile       = codegenDir / "generated.json"
+  val publishedLocalFile  = publishLocalDir / "published.json"
+  val publishedMavenFile  = publishMavenDir / "published.json"
+  val publishedGithubFile = publishGithubDir / "published.json"
+
+  def publishOpts(heapMaxGb: Int = 32, jarCompression: Int = 1, sources: Boolean = false, docs: Boolean = false): Vector[os.Shellable] =
```

Comment:
why jarCompression = 1? isn't this the lowest compression?

Reply:
ah, because the default is for local? but then you have ifs below so maybe no need for this default at all? it's risky, 9 would be better

Reply:
default in scala-cli AFAIR is -1 (no compression), I've enabled highest because our package jars tend to be huge ;()